hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/DeleteAllowanceChecks.java

```diff
@@ -0,0 +1,352 @@
+			final var token = tokenStore.loadPossiblyPausedToken(Id.fromGrpcToken(tokenId));
+
+			if (token.isFungibleCommon()) {
+				return FUNGIBLE_TOKEN_IN_NFT_ALLOWANCES;
+			}
+
+			final var fetchResult = fetchOwnerAccount(owner, payerAccount, accountStore);
+			if (fetchResult.getRight() != OK) {
+				return fetchResult.getRight();
+			}
+			final var ownerAccount = fetchResult.getLeft();
+
+			if (!ownerAccount.isAssociatedWith(Id.fromGrpcToken(tokenId))) {
+				return TOKEN_NOT_ASSOCIATED_TO_ACCOUNT;
+			}
+			final var validity = validateSerialNums(serialNums, ownerAccount, token);
+			if (validity != OK) {
+				return validity;
+			}
+		}
+
+		return OK;
+	}
+
+	/**
+	 * Checks if the nft allowances are repeated. Also validates if the serial numbers are repeated.
+	 *
+	 * @param nftAllowances
+	 * @return
+	 */
+	boolean repeatedAllowances(final List<NftRemoveAllowance> nftAllowances) {
+		Map<Pair<AccountID, TokenID>, List<Long>> seenNfts = new HashMap<>();
+		for (var allowance : nftAllowances) {
+			final var key = Pair.of(allowance.getOwner(), allowance.getTokenId());
+			if (seenNfts.containsKey(key)) {
+				if (serialsRepeated(seenNfts.get(key), allowance.getSerialNumbersList())) {
+					return true;
+				} else {
+					final var list = new ArrayList<Long>();
+					list.addAll(seenNfts.get(key));
+					list.addAll(allowance.getSerialNumbersList());
+					seenNfts.put(key, list);
+				}
+			} else {
+				seenNfts.put(key, allowance.getSerialNumbersList());
+			}
+		}
+		return false;
+	}
+
+	boolean serialsRepeated(final List<Long> existingSerials, final List<Long> newSerials) {
+		for (var serial : newSerials) {
+			if (existingSerials.contains(serial)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	ResponseCodeEnum validateSerialNums(final List<Long> serialNums,
+			final Account ownerAccount,
+			final Token token) {
+		if (hasRepeatedSerials(serialNums)) {
+			return REPEATED_SERIAL_NUMS_IN_NFT_ALLOWANCES;
+		}
+
+		if (serialNums.isEmpty()) {
+			return EMPTY_ALLOWANCES;
+		}
+
+		for (var serial : serialNums) {
+			final var nftId = NftId.withDefaultShardRealm(token.getId().num(), serial);
+
+			if (serial < 0 || serial == 0 || !nftsMap.get().containsKey(EntityNumPair.fromNftId(nftId))) {
+				return INVALID_TOKEN_NFT_SERIAL_NUMBER;
+			}
+
+			final var nft = nftsMap.get().get(EntityNumPair.fromNftId(nftId));
+			if (!AllowanceChecks.validOwner(nft, ownerAccount, token)) {
+				return SENDER_DOES_NOT_OWN_NFT_SERIAL_NO;
+			}
+		}
+
+		return OK;
+	}
+
+	ResponseCodeEnum commonChecks(
+			final List<CryptoRemoveAllowance> cryptoAllowances,
+			final List<TokenRemoveAllowance> tokenAllowances,
+			final List<NftRemoveAllowance> nftAllowances) {
+		// each serial number of an NFT is considered as an allowance.
+		// So for Nft allowances aggregated amount is considered for limit calculation
+		final var totalAllowances = cryptoAllowances.size() + tokenAllowances.size() +
```

Comment:
Don't only `NftAllowances` with `approved_for_all = true` count against the `MAX_ALLOWANCES_EXCEEDED` limit now?

Reply:
@tinker-michaelj  True. But, that limit is for number of allowances on account. We limit the total allowances, including serials on transaction though. @superboo  @anighanta Please correct me if I am wrong.